fvm/transactionInvoker.go

```diff
@@ -181,69 +206,56 @@ func (i TransactionInvoker) normalExecution(
 	// disable the computation/memory limit checks on storage checks,
 	// so we don't error from computation/memory limits on this part.
 	// We cannot charge the user for this part, since fee deduction already happened.
-	txnState.RunWithAllLimitsDisabled(func() {
-		err = NewTransactionStorageLimiter().CheckLimits(env, txnState.UpdatedAddresses())
+	executor.txnState.RunWithAllLimitsDisabled(func() {
+		err = NewTransactionStorageLimiter().CheckLimits(
+			executor.env,
+			executor.txnState.UpdatedAddresses())
 	})
 
 	return
 }
 
 // Clear changes and try to deduct fees again.
-func (i TransactionInvoker) errorExecution(
-	proc *TransactionProcedure,
-	txnState *state.TransactionState,
-	nestedTxnId state.NestedTransactionId,
-	env environment.Environment,
-	errs *errors.ErrorsCollector,
-) error {
-	txnState.DisableAllLimitEnforcements()
-	defer txnState.EnableAllLimitEnforcements()
+func (executor *invocationExecutor) errorExecution() {
```

Comment:
note: this no longer returns the fatal error.  `errs.ErrorOrNil()` is called in `Process` instead (see line 103)